### PR TITLE
Chapter 2 answers

### PR DESCRIPTION
**Pull request by SandyThomson**
At the time of writing, only Exercise 1 done.

### CODE REVIEW

File: chapter02/sdt/ex1.pl [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+use warnings;
+use strict;
+use Cwd qw( abs_path );
+use File::Spec qw( curdir );
+
+opendir(DIR, File::Spec->curdir()) or die $!;
+while (my $file = readdir(DIR)) {
+  print " " x 4 .abs_path($file) . "\n";
+}
+closedir(DIR);
```
Comment (alexdeas) [inline chapter02/sdt/ex1.pl:11]: Normally in a review I'd start talking about file handle references here, but those come later in the book so I'll avoid spoilers.

File: chapter02/sdt/ex1.pl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/perl
+use warnings;
+use strict;
+use Cwd qw( abs_path );
+use File::Spec qw( curdir );
+
+opendir(DIR, File::Spec->curdir()) or die $!;
```
Comment (alexdeas) [inline chapter02/sdt/ex1.pl:7]: `curdir` seems useful. Probably for this case `"."` was enough, but modules!

Using opendir would also give you hidden files, including `.` and `..`.

File: chapter02/sdt/output.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+    /home/sandy/Intermediate-Perl/chapter02/sdt/output.txt
+    /home/sandy/Intermediate-Perl/chapter02
+    /home/sandy/Intermediate-Perl/chapter02/sdt/ex1.pl
+    /home/sandy/Intermediate-Perl/chapter02/sdt
```
Comment (dickyince) [inline chapter02/sdt/output.txt:4]: check out all the files

File: chapter02/sdt/ex1.pl [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/perl
+use warnings;
+use strict;
+use Cwd qw( abs_path );
+use File::Spec qw( curdir );
```
Comment (dickyince) [inline chapter02/sdt/ex1.pl:5]: use all the modules!

**Pull request merged**